airflow_providers_wherobots/wherobots/models.py

```diff
@@ -0,0 +1,106 @@
+"""
+The data models for the WhereRobots API
+"""
+
+import string
+from datetime import datetime
+from enum import auto, Enum
+from typing import Optional, Sequence
+
+from pydantic import BaseModel, Field, ConfigDict, computed_field
+from strenum import StrEnum
+from wherobots.db import Runtime
+
+RUN_NAME_CHARSET = string.ascii_letters + string.digits + "-_."
+
+
+class RunStatus(str, Enum):
+    PENDING = auto()
+    RUNNING = auto()
+    FAILED = auto()
+    COMPLETED = auto()
+    CANCELLED = auto()
+
+    def is_active(self) -> bool:
+        return self in [self.PENDING, self.RUNNING]
+
+
+class WherobotsModel(BaseModel):
+    ext_id: str = Field(alias="id")
+    create_time: datetime = Field(alias="createTime")
+    update_time: datetime = Field(alias="updateTime")
+    model_config = ConfigDict(from_attributes=True, populate_by_name=True)
+
+
+class Run(WherobotsModel):
+    name: str
+    status: RunStatus
+    start_time: Optional[datetime] = Field(default=None, alias="startTime")
+    end_time: Optional[datetime] = Field(default=None, alias="completeTime")
+
+
+class PythonRunPayload(BaseModel):
+    """
+    Model for the payload of Run with type == "python"
+    """
+
+    # For airflow to render the template fields
+    template_fields: Sequence[str] = Field(
+        ("uri", "args", "entrypoint"), exclude=True, init=False
+    )
+
+    uri: str = Field(..., alias="uri")
+    args: list[str] = []
+    entrypoint: Optional[str] = Field(None, alias="entrypoint")
+
+    @classmethod
+    def create(cls, uri: str, args: list[str], entrypoint: Optional[str] = None):
+        return cls(uri=uri, args=args, entrypoint=entrypoint)
```

Comment:
Why are those `create()` methods required? Can't we just do:

```python
PythonRunPayload(uri=uri, args=args, entrypoint=entrypoint)
```

?

Reply:
The original Pydnatic models doesn't come with hints, I add these methods so that users can simply do `Model.create` with the hint available in their IDE. WDYT

Reply:
Ah I see. I don't have a strong opinion on this, so it's okay.